Pipe API formatters need to be public

Compile error in downstream project

https://github.com/heuermh/adam-snpeff/blob/master/src/main/scala/com/github/heuermh/adam/snpeff/AdamSnpEff.scala#L56

```
[INFO] Compiling 1 Scala source to /Users/heuermh/working/adam-snpeff/target/scala-2.10.5/classes...
[ERROR] /Users/heuermh/working/adam-snpeff/src/main/scala/com/github/heuermh/adam/snpeff/AdamSnpEff.scala:56: class VCFInFormatter in package variant cannot be accessed in package org.bdgenomics.adam.rdd.variant
[ERROR]     val output: VariantContextRDD = input.pipe[VariantContext, VariantContextRDD, VCFInFormatter](snpEffCommand)
[ERROR]                                                                                   ^
[ERROR] one error found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala
@@ -55,7 +55,7 @@ object InterleavedFASTQInFormatter extends InFormatterCompanion[Fragment, Fragme
   }
 }
 
-private[fragment] class InterleavedFASTQInFormatter private (
+class InterleavedFASTQInFormatter private (
     conf: Configuration) extends InFormatter[Fragment, FragmentRDD, InterleavedFASTQInFormatter] with Logging {
 
   protected val companion = InterleavedFASTQInFormatter

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/BAMInFormatter.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/BAMInFormatter.scala
@@ -40,7 +40,7 @@ object BAMInFormatter extends AnySAMInFormatterCompanion[BAMInFormatter] {
   }
 }
 
-private[read] case class BAMInFormatter private (
+case class BAMInFormatter private (
     header: SAMFileHeaderWritable,
     recordGroups: RecordGroupDictionary,
     converter: AlignmentRecordConverter) extends AnySAMInFormatter[BAMInFormatter] {

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/SAMInFormatter.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/SAMInFormatter.scala
@@ -40,7 +40,7 @@ object SAMInFormatter extends AnySAMInFormatterCompanion[SAMInFormatter] {
   }
 }
 
-private[read] case class SAMInFormatter private (
+case class SAMInFormatter private (
     header: SAMFileHeaderWritable,
     recordGroups: RecordGroupDictionary,
     converter: AlignmentRecordConverter) extends AnySAMInFormatter[SAMInFormatter] {

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/variant/VCFInFormatter.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/variant/VCFInFormatter.scala
@@ -51,7 +51,7 @@ object VCFInFormatter extends InFormatterCompanion[VariantContext, VariantContex
   }
 }
 
-private[variant] case class VCFInFormatter private (
+case class VCFInFormatter private (
     sequences: SequenceDictionary,
     samples: Seq[String],
     headerLines: Seq[VCFHeaderLine]) extends InFormatter[VariantContext, VariantContextRDD, VCFInFormatter] {

